controllers/k6_finish.go

```diff
@@ -20,52 +18,14 @@ import (
 func FinishJobs(ctx context.Context, log logr.Logger, k6 *v1alpha1.K6, r *K6Reconciler) (ctrl.Result, error) {
 	log.Info("Waiting for pods to finish")
 
-	// Here we assume that the test runs for some time and there is no need to
-	// check it more often than twice in a minute.
-	//
-	// The total timeout for the test is set to duration of the test + 2 min.
-	// These 2 min are meant to cover the time needed to start the pods: sometimes
-	// pods are ready a bit later than operator reaches this stage so from the
-	// viewpoint of operator it takes longer. This behaviour depends on the setup of
-	// cluster. 2 min are meant to be a sufficient safeguard for such cases.
-
-	testDuration := inspectOutput.TotalDuration.TimeDuration()
-
-	err := wait.PollImmediate(time.Second*30, testDuration+time.Minute*2, func() (done bool, err error) {
-		selector := labels.SelectorFromSet(map[string]string{
-			"app":    "k6",
-			"k6_cr":  k6.Name,
-			"runner": "true",
-		})
-
-		opts := &client.ListOptions{LabelSelector: selector, Namespace: k6.Namespace}
-		jl := &batchv1.JobList{}
-
-		if err := r.List(ctx, jl, opts); err != nil {
-			log.Error(err, "Could not list jobs")
-			return false, nil
-		}
-
-		// TODO: We should distinguish between Suceeded/Failed/Unknown
-		var finished int32
-		for _, job := range jl.Items {
-			if job.Status.Active != 0 {
-				continue
-			}
-			finished++
-		}
+	allK6PodsAreFinalized, err := allk6RunnerPodsAreFinished(ctx, log, k6, r)
```

Comment:
The same as with start and initialize: let's keep previous structure.

Reply:
Code removed